src/main/java/com/SAS/teamManagenemt/TeamManagement.java

```diff
@@ -77,8 +82,55 @@ else if (user instanceof TeamManager && ((TeamManager)user).getTeam() == team) {
      * @return true - if the user is the team owner, otherwise - false
      */
     private boolean ownsTeam(Team team, User nominatedBy) {
-         return nominatedBy instanceof TeamOwner && ((TeamOwner) nominatedBy).getTeam() == team;
+        return nominatedBy instanceof TeamOwner && ((TeamOwner) nominatedBy).getTeam() == team;
     }
 
+    /**
+     * The fcuntion receives the parameters of new transaction and the team and add it if it's legal -
+     * not exceed the budget of the team
+     * @param team
+     * @param transDetails
+     * @return
+     */
+    public Transaction addTransactionToTeam(Team team, List<String> transDetails, User teamOwner) {
+        if (ownsTeam(team, teamOwner)) {
+            //first amount, second type, third date and last description
+            double amount = Double.parseDouble(transDetails.get(0));
+            TransactionType type = convertStringToTrasactionType(transDetails.get(1));
+            LocalDate date = LocalDate.parse(transDetails.get(2));
+            String description = transDetails.get(3);
+
+            //create the transaction
+            Transaction transaction = new Transaction(amount, type, date, team, description, (TeamOwner) teamOwner);
+
+            boolean isTransAdded = team.addTransactionToTeam(transaction);
+            if (!isTransAdded) {
+                System.out.println("The transaction could not be added to the team, it exceeded the team's budget.");
+            }
+
+            return transaction;
```

Comment:
Maybe we need to make the return different params if addTransaction succeeded or not.
The difference is only the sout and it will be hard to check the method.

Reply:
You're right, changed it to boolean.